### DISCUSSION TITLE
Proper flush buffer without reset?

### DISCUSSION THREAD

**Discussion by MadTooler** [grblHAL/RP2040#55] [Q&A] [answered]
I am working on a custom sender feature.  At times, I need to feed hold the gcode/grbl, get status for line number of gcode currently in hold, then cancel or stop all buffered gcode.  Pretty much like what the jog cancel does for jogging.

I have been using soft reset, but I haven't gotten some of the handshaking to work well and get an alarm3 sometimes when I send the reset.  

Is there a proper way to cancel the current gcode that is in a hold and clear the buffer?

**Comment by terjeio** [comment]
Send jog cancel (0x85) - this clears both the serial input buffer and the line buffer.

**Reply by terjeio** [reply]
> Is there a proper way to cancel the current gcode that is in a hold and clear the buffer?

Send a stop command (0x19) instead of reset after the jog cancel (or before - test what works)?

**Reply by MadTooler** [reply]
I should get my version of the firmware updated.  I do not know if any of my experiences would be affected by the changes you have made since last November.  My custom machine settings are the only reason I am behind.

It looks like 0x19 stop works to stop it and I think it clears the buffer as after I send it I see `<Idle|MPos:-714.987,4.046,0.000|Bf:35,1023|FS:0,0|A:>`

BTW, I could not find the stop command anywhere in the command documentation.  I do see it in the firmware, now that I know what to look for, but I look to the documentation before I start digging through the code since a lot of it is more technical than my current comprehension.

Jog cancel only works on jogs, does not stop gcode with the version I am running.  It also does not free a hold.  Depending on how the grbl 1.1 protocol is interpreted, that may or may not be as intended.  When the system is in a feed HOLD state, it also does not cancel nor seem to flush the buffer if filled with gcode.  However, it does work well for any buffered jog commands.

Thank you again!

**Comment by MadTooler** [comment, marked as answer]
> Send a stop command (0x19) instead of reset after the jog cancel (or before - test what works)?

I tried to mark the stop command as the answer, but github took away the option.  Reposting it to close this as solved.
